src/background/Wallet/WalletRecord.ts

```diff
@@ -260,6 +276,44 @@ export class WalletRecordModel {
       if (!draft.feed.dismissedAbilities) {
         draft.feed.dismissedAbilities = [];
       }
+      /**
+       * NOTE:
+       * (De)duplication logic:
+       * When adding signer/hw wallet, existing readonly wallet should be removed
+       * When adding readonly address, it should be ignored if signer/hw wallet exists
+       * Duplication of addresses between mnemonic/privateKey/HW wallets is allowed
+       */
+
+      // Reuse existing wallet names for new addresses
+      for (const wallet of walletContainer.wallets) {
```

Comment:
Should we be able to add watch address that is already existed as imported?

Reply:
No, that's the only group for which duplication isn't allowed :) When you add existing address as a watch address, the record is not updated except for making the new address a current one.